packages/gatsby/cache-dir/ensure-resources.js

```diff
@@ -131,8 +131,13 @@ class EnsureResources extends React.Component {
     if (!this.hasResources(this.state.pageResources) && isInitialRender) {
       window.___failedResources = true
 
-      // prevent hydrating
-      throw new Error(`Missing resources for ${this.state.location.pathname}`)
+      console.error(`Missing resources for ${this.state.location.pathname}`)
+      return (
+        <div
+          dangerouslySetInnerHTML={{ __html: `` }}
+          suppressHydrationWarning
+        />
+      )
```

Comment:
@davidbailey00 I would like sanity check here:
this seems to do similar thing (preventing react from cleaning up `___gatsby` div) and it works in more scenarios than throwing (in case we throw before any parent component "renders" any DOM element, react will wipe `___gatsby` div). It seems like we were "lucky" with this, because <div> was always there coming from <Router> above <EnsureResources>. My change moves <Router> below so no DOM element is above, which was causing issues.

Reply:
this means we will do a full client rehydration if something goes wrong, right? maybe add a comment about it?

Reply:
It's partial hydration - things above the actual page will be hydrated - so if users use `wrapRootElement`/`wrapPageElement` or `gatsby-plugin-layout` those will be actually hydrated, but the content of the page itself (in case template chunk or query results can't be loaded) won't be trully hydrated, but content from static html will be there

This is a neat trick that you can do when using:
```
<div
          dangerouslySetInnerHTML={{ __html: `` }}
          suppressHydrationWarning
        />
```
during hydration, as it won't try to actually change DOM nodes under it.

But yeah, I should add more context there for sure